test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -0,0 +1,995 @@
+    expect(aliceAddress).toBe(alice._address);
+
+    await hardhat.fork();
+
+    await hardhat.fund(alice._address, [
+      parseAmount('8000000', USDC_MAINNET),
+      parseAmount('5000000', USDT_MAINNET),
+      parseAmount('1000', UNI_MAINNET),
+      parseAmount('5000000', DAI_MAINNET),
+    ], [
+      "0x47ac0fb4f2d84898e4d9e7b4dab3c24507a6d503", // Binance peg tokens
+    ])
+
+    await hardhat.fund(alice._address, [
+      parseAmount('4000', WETH9[1]),
+    ], [
+      "0x6555e1CC97d3cbA6eAddebBCD7Ca51d75771e0B8", // WETH token
+    ])
+
+    // alice should always have 10000 ETH
+    const aliceEthBalance = await hardhat.provider.getBalance(alice._address);
+    expect(aliceEthBalance).toEqual(parseEther('10000'));
+    const aliceUSDCBalance = await hardhat.getBalance(alice._address, USDC_MAINNET);
+    expect(aliceUSDCBalance).toEqual(parseAmount('8000000', USDC_MAINNET));
+    const aliceUSDTBalance = await hardhat.getBalance(alice._address, USDT_MAINNET);
+    expect(aliceUSDTBalance).toEqual(parseAmount('5000000', USDT_MAINNET));
+    const aliceWETH9Balance = await hardhat.getBalance(alice._address, WETH9[1]);
+    expect(aliceWETH9Balance).toEqual(parseAmount('4000', WETH9[1]));
+    const aliceDAIBalance = await hardhat.getBalance(alice._address, DAI_MAINNET);
+    expect(aliceDAIBalance).toEqual(parseAmount('5000000', DAI_MAINNET));
+    const aliceUNIBalance = await hardhat.getBalance(alice._address, UNI_MAINNET);
+    expect(aliceUNIBalance).toEqual(parseAmount('1000', UNI_MAINNET));
+
+    alphaRouter = new AlphaRouter({
+      chainId: ChainId.MAINNET,
+      provider: hardhat.providers[0]!,
+      multicall2Provider
+    })
+  })
+
+  /**
+   *  tests are 1:1 with routing api integ tests
+   */
+  for (const tradeType of [TradeType.EXACT_INPUT, TradeType.EXACT_OUTPUT]) {
+    describe(`${ID_TO_NETWORK_NAME(1)} alpha - ${tradeType}`, () => {
+      describe(`+ simulate swap`, () => {
+        it('erc20 -> erc20', async () => {
+          // declaring these to reduce confusion
+          const tokenIn = USDC_MAINNET;
+          const tokenOut = USDT_MAINNET;
+          const amount = tradeType == TradeType.EXACT_INPUT ?
+            parseAmount('100', tokenIn)
+            : parseAmount('100', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG
+            }
+          );
+
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quote,
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            methodParameters,
+          } = convertSwapDataToResponse(amount, tradeType, swap!)
```

Comment:
most of the work in `convertSwapDataToResponse` is parsing the route. since we don't check the route here at all would either suggest creating a different function that parses quote, quoteDecimals, quoteGasAdjustedDecimals, and methodParams or just parsing it directly from swap

Reply:
would even suggest just converting `convertSwapDataToResponse` into a helper that specifically does route parsing since we don't check a lot of the other fields

Reply:
Sounds good, I have one for route parsing and one for quote now